crates/core/tedge_agent/src/agent.rs

```diff
@@ -284,6 +298,39 @@ impl Agent {
         )
         .await?;
 
+        // TODO: replace with a call to entity store when we stop assuming default MQTT schema
+        let is_main_device =
+            self.config.mqtt_device_topic_id == EntityTopicId::default_main_device();
+        if is_main_device {
+            info!("Running as a main device, starting tedge_to_te_converter");
+
+            runtime.spawn(tedge_to_te_converter).await?;
+        } else {
+            info!("Running as a child device, tedge_to_te_converter disabled");
+        }
+
+        if *entity_topic_id
+            .default_parent_identifier()
+            .unwrap()
+            .as_str()
+            == *self.config.fts_entity
+        {
```

Comment:
I don't understand the need motivating this change.

Having the FTS service running on the main device sounds to be a default that makes.
And even if more flexibility might help in some case, this is independent of this PR.

Reply:
As agreed in the call, this setting was removed and File Transfer Service was reverted to spawn only on the main device.